### PR TITLE
Added type definitions

### PR DESCRIPTION
**Pull request by ocarlott**
I added type definitions for TypeScript users.

### CODE REVIEW

**Comment by MiLeung** [comment]
I'm probably not going to make this package use TypeScript because I don't have the knowledge about ts to make an informed decision of whether or not to merge it. It would be nice if others could chime in on whether or not this PR should be merged.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    ref: MutableRefObject<SpriteSheet | null>,
+    animations: AnimationConfig, // see example
+    viewStyle?: StyleProp<ViewStyle>, // styles for the sprite sheet container
+    imageStyle?: StyleProp<ImageStyle>, // styles for the sprite sheet
+    height?: number, // set either height, width, or none,
+    width?: number, // but not both height and width
+    onLoad?: () => void
+}
+
+export class SpriteSheet extends PureComponent<RequireOnlyOne<ISpriteSheetProps, "height" | "width">, {}> {
```
Comment (JeffGuKang) [inline index.d.ts]: It cause ts error as below when `width` and `height` are undefined. 

```
Property 'width' is missing in type '{ ref: RefObject<SpriteSheet>; source: number; columns: number; rows: number; animations: { basic: number[]; }; }' but required in type 'Readonly<Pick<SpriteSheetProps, "source" | "columns" | "rows" | "ref" | "animations" | "viewStyle" | "imageStyle" | "onLoad"> & Required<Pick<SpriteSheetProps, "width">> & Partial<...>>'.ts(2769)
```

`height` and `width` can be undefined both.

Comment (JeffGuKang) [reply index.d.ts]: I think `export default class SpriteSheet extends PureComponent<SpriteSheetProps, {}> {` is enough. It is not hard to find how to use `width` or `height` in documents(README).

**Comment by JeffGuKang** [comment]
This PR is useful for me. I applied it to my project with this module and checked it almost worked well.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    source: number, // source must be required; { uri } will not work
+    columns: number,
+    rows: number,
+    ref: MutableRefObject<SpriteSheet | null>,
+    animations: AnimationConfig, // see example
+    viewStyle?: StyleProp<ViewStyle>, // styles for the sprite sheet container
+    imageStyle?: StyleProp<ImageStyle>, // styles for the sprite sheet
+    height?: number, // set either height, width, or none,
+    width?: number, // but not both height and width
+    onLoad?: () => void
```
Comment (JeffGuKang) [inline index.d.ts]: Need `onFinish?: () => void`

File: index.d.ts (hunk 1 above)
Comment (ocarlott) [reply index.d.ts]: You'r right. I have fixed that. Thanks 👍

**Comment by MiLeung** [comment]
@JeffGuKang almost worked well? Will this pr work with plain JS React Native projects?

**Comment by JeffGuKang** [comment]
> @JeffGuKang almost worked well? Will this pr work with plain JS React Native projects?

Yup. `index.t.ds` has no effect on plain JS project. 

https://stackoverflow.com/questions/21247278/about-d-ts-in-typescript

**Comment by MiLeung** [comment]
Thanks, published these changes in v1.1.6

**Pull request merged**